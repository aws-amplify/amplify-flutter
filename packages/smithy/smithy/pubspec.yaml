name: smithy
description: Smithy client runtime for Dart with common utilities for I/O and serialization.
version: 0.5.2
homepage: https://github.com/aws-amplify/amplify-flutter/tree/main
repository: https://github.com/aws-amplify/amplify-flutter/tree/main/packages/smithy/smithy
issue_tracker: https://github.com/aws-amplify/amplify-flutter/issues

environment:
  sdk: ^3.0.0

dependencies:
  async: ^2.10.0
  aws_common: ">=0.6.0 <0.7.0"
  built_collection: ^5.0.0
  built_value: ">=8.6.0 <8.7.0"
  collection: ^1.15.0
  convert: ^3.0.0
  crypto: ^3.0.0
  fixnum: ^1.0.0
  http_parser: ^4.0.0
  intl: ">=0.18.0 <1.0.0"
  json_annotation: ">=4.8.1 <4.9.0"
  meta: ^1.7.0
  path: ^1.8.0
  retry: ^3.1.0
  shelf: ^1.1.0
  typed_data: ^1.3.0
  xml: 6.4.2

dev_dependencies:
  amplify_lints: ">=3.0.0 <3.1.0"
  build_runner: ^2.4.0
  built_value_generator: 8.6.1
  json_serializable: 6.7.1
  stack_trace: ^1.10.0
  test: ^1.22.1
