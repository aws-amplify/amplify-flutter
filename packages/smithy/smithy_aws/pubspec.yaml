name: smithy_aws
description: Smithy runtime for AWS clients with utilities for endpoint resolution, retry behavior, and SigV4 signing.
version: 0.6.0
homepage: https://github.com/aws-amplify/amplify-flutter/tree/main
repository: https://github.com/aws-amplify/amplify-flutter/tree/main/packages/smithy/smithy_aws
issue_tracker: https://github.com/aws-amplify/amplify-flutter/issues

environment:
  sdk: ^3.0.0

dependencies:
  aws_common: ">=0.6.0 <0.7.0"
  aws_signature_v4: ">=0.5.0 <0.6.0"
  built_collection: ^5.0.0
  built_value: ">=8.6.0 <8.8.0"
  collection: ^1.15.0
  convert: ^3.0.0
  crclib: ^3.0.0
  crypto: ^3.0.0
  intl: ">=0.18.0 <1.0.0"
  json_annotation: ">=4.8.1 <4.9.0"
  meta: ^1.7.0
  path: ^1.8.0
  smithy: ">=0.6.0 <0.7.0"
  xml: 6.3.0

dev_dependencies:
  amplify_lints: ">=3.0.0 <3.1.0"
  build_runner: ^2.4.0
  built_value_generator: 8.7.0
  file: ">=6.0.0 <8.0.0"
  glob: ^2.0.2
  json_serializable: 6.7.1
  pubspec_parse: ^1.2.0
  test: ^1.22.1
  yaml: ^3.1.0
