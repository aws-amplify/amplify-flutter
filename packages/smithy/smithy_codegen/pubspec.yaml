name: smithy_codegen
description: Smithy model generator from AST representation
version: 0.3.1
repository: https://github.com/aws-amplify/amplify-flutter/tree/main/packages/smithy/smithy_codegen
publish_to: none

environment:
  sdk: ^3.0.0

dependencies:
  args: ^2.3.0
  aws_common: any
  aws_signature_v4: any
  build_cli_annotations: ^2.0.0
  built_collection: ^5.1.1
  built_value: ">=8.6.0 <8.8.0"
  code_builder: 4.7.0
  collection: ^1.15.0
  crclib: ^3.0.0
  dart_style: 2.3.2
  fixnum: ^1.0.0
  grpc: ^3.0.2
  html2md: ^1.2.5
  jmespath: ^2.0.0
  json_annotation: ">=4.8.1 <4.9.0"
  meta: ^1.7.0
  path: ^1.8.0
  protobuf: ^2.0.1
  pub_semver: ^2.1.0
  pubspec_parse: ^1.2.0
  shelf: ^1.1.0
  shelf_router: ^1.1.0
  smithy: ">=0.6.0 <0.7.0"
  smithy_aws: ">=0.6.0 <0.7.0"
  tuple: ^2.0.0
  xml: 6.3.0
  yaml_edit: ^2.0.1

dev_dependencies:
  amplify_lints: ">=3.0.0 <3.1.0"
  build_cli: ^2.1.6
  build_runner: ^2.4.0
  build_verify: ^3.0.0
  build_version: ^2.1.0
  built_value_generator: 8.8.0
  json_serializable: 6.7.1
  smithy_test:
    path: ../smithy_test
  test: ^1.22.1

executables:
  smithy_codegen:
