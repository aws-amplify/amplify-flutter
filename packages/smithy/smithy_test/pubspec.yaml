name: smithy_test
description: Test utilities for Smithy packages.
version: 0.1.0
repository: https://github.com/aws-amplify/amplify-flutter/tree/main/packages/smithy/smithy_test
publish_to: none

environment:
  sdk: ^3.0.0

dependencies:
  async: ^2.10.0
  aws_common: any
  aws_signature_v4: any
  built_value: ">=8.6.0 <8.8.0"
  collection: ^1.15.0
  fixnum: ^1.0.0
  path: ^1.8.0
  smithy: any
  test: ^1.22.1
  tuple: ^2.0.0
  xml: 6.3.0

dev_dependencies:
  amplify_lints:
    path: ../../amplify_lints
