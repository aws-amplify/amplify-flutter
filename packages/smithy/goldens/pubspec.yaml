name: goldens
description: Golden files for smithy-dart
version: 1.0.0
publish_to: none

environment:
  sdk: ">=2.17.0 <3.0.0"

dependencies:
  aws_common: ">=0.4.0 <0.5.0"
  smithy: ">=0.4.0+1 <0.5.0"

dependency_overrides: 
  aws_common:
    path: ../../aws_common
  aws_signature_v4:
    path: ../../aws_signature_v4
  smithy:
    path: ../smithy
  smithy_aws:
    path: ../smithy_aws
  smithy_codegen:
    path: ../smithy_codegen

dev_dependencies:
  args: ^2.3.1
  file: ^6.1.2
  glob: ^2.0.2
  lints: ^2.0.0
  path: any
  pubspec_parse: ^1.2.0
  smithy_codegen:
    path: ../smithy_codegen
  test: ^1.16.0
